opm/simulators/wells/StandardWellEval.cpp

```diff
@@ -950,13 +950,14 @@ computeConnectionDensities(const std::vector<double>& perfComponentRates,
             if (!rvmax_perf.empty() && mix[gaspos] > 1e-12) {
                 rv = std::min(mix[oilpos]/mix[gaspos], rvmax_perf[perf]);
             }
-            if (rs != 0.0) {
+            double d = 1.0 - rs*rv;
+            if (rs > 0.0 && d > 0.0) {
                 // Subtract gas in oil from gas mixture
-                x[gaspos] = (mix[gaspos] - mix[oilpos]*rs)/(1.0 - rs*rv);
+                x[gaspos] = (mix[gaspos] - mix[oilpos]*rs)/d;
             }
-            if (rv != 0.0) {
+            if (rv > 0.0 && d > 0.0) {
                 // Subtract oil in gas from oil mixture
-                x[oilpos] = (mix[oilpos] - mix[gaspos]*rv)/(1.0 - rs*rv);;
+                x[oilpos] = (mix[oilpos] - mix[gaspos]*rv)/d;
             }
```

Comment:
We should have debugging output here, for the abnormal d values (d <= 0.0 || d > 1.0). 

And I do not think we should use the `d` values that bigger than 1 either, which implies negative `rs` or `rv`.

Reply:
I think the correct solution for d > 1 is to set it to 1. If rs > 0.0 and rv < 0.0 We should remove the effect of rs as if there is no rv. An alternative is to force rs/rv >= 0